libgraphics/include/pivot/graphics/vk_utils.hxx

```diff
@@ -29,7 +31,31 @@ constexpr void vk_try(vk::Result err)
 constexpr void vk_try(VkResult res) { vk_try(vk::Result(res)); }
 
 /// Read a whole file into a vector of byte.
-std::vector<std::byte> readFile(const std::string &filename);
+template <typename T = std::byte>
+std::vector<T> readBinaryFile(const std::filesystem::path &filename)
+{
+    size_t fileSize = std::filesystem::file_size(filename);
+    std::ifstream file(filename, std::ios::binary);
+    std::vector<T> fileContent(fileSize / sizeof(T));
+
+    if (!file.is_open()) throw std::runtime_error("failed to open file " + filename.string());
+    file.read(reinterpret_cast<char *>(fileContent.data()), fileSize * sizeof(T));
```

Comment:
Check how many characters were read with `gcount`, and check that no error occured with `fail` and `bad`.